### ISSUE TITLE
Allow a callback to be registered to fire only N times, instead of always

### ISSUE THREAD

**Issue by PatSmuk** [ElunaLuaEngine/Eluna#128]
**TL;DR:** Make RegisterXEvent functions more like [WorldObject:RegisterEvent](http://eluna.emudevs.com/WorldObject/RegisterEvent.html) and have a `repeats` parameter to control how many times the callback will be called.
## Current Situation

There are a number of functions to register callbacks for events, all named RegisterXEvent (where X is a class, e.g. [RegisterCreatureEvent](http://eluna.emudevs.com/Global/RegisterCreatureEvent.html)). However, there is no way to control how many times the callback is called, or to unregister a callback.

There are also the functions [CreateLuaEvent](http://eluna.emudevs.com/Global/CreateLuaEvent.html) and [WorldObject:RegisterEvent](http://eluna.emudevs.com/WorldObject/RegisterEvent.html) which allow you to register a callback and specify how many times it will be called, but these only work for timers.

This means that adding and removing callbacks on the fly for anything but timers is currently impossible and limits the flexibility of Eluna.
## Proposal

Add a new parameter to all RegisterXEvent functions, `repeats`, to control how many times the callback will fire.

If `repeats` is 0 or only 3 parameters are supplied the functions will work as before (always call the callback), for consistency with WorldObject:RegisterEvent and backwards compatibility
